src/saltext/vmware/states/tag.py

```diff
@@ -0,0 +1,141 @@
+# SPDX-License-Identifier: Apache-2.0
+import logging
+
+import saltext.vmware.utils.common as utils_common
+import saltext.vmware.utils.connect as connect
+import saltext.vmware.utils.vm as utils_vm
+from saltext.vmware.modules.tag import update
+
+log = logging.getLogger(__name__)
+
+try:
+    from pyVmomi import vim
+
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+
+
+__virtualname__ = "vmware_tag"
+__proxyenabled__ = ["vmware_tag"]
+
+
+def __virtual__():
+    if not HAS_PYVMOMI:
+        return False, "Unable to import pyVmomi module."
+    return __virtualname__
+
+
+def present(name, description=None, category_id=None):
+    """
+    Create and update a tag instance.
+
+    name
+        Name of tag.
+
+    description
+        Description of tag.
+
+    category_id
+        (string) Category ID of type: com.vmware.cis.tagging.Tag. Only optional when updating
+    """
+    ret = {"name": name, "changes": {}, "result": True, "comment": ""}
+    res = connect.request(
+        "/rest/com/vmware/cis/tagging/tag", "GET", opts=__opts__, pillar=__pillar__
+    )
+    response = res["response"].json()
+    token = res["token"]
+    found = None
+    for tag in response["value"]:
+        url = f"/rest/com/vmware/cis/tagging/tag/id:{tag}"
+        tag_ref = connect.request(url, "GET", token=token, opts=__opts__, pillar=__pillar__)
+        tag_ref = tag_ref["response"].json()
+        if tag_ref["value"]["name"] == name:
+            found = tag_ref["value"]
+            break
+    if found:
```

Comment:
Need to handle dry run mode (state.apply with test=True).

```
        if __opts__["test"]:
               .....
                ret["comment"] = "tag - {} will be created on resource {}..something like that.."
                ret["result"] = None
       elif ....
```

same for absent.